### ISSUE TITLE
Azure login doesn't work on Azure China

### ISSUE THREAD

**Issue by catweisun** [Azure/azure-xplat-cli#2643]
CLI Version: **0.9.16**
OS Type:  Win
Installation via:   win installer

Mode: **ARM / ASM**

Description:
`azure login -e AzureChinaCloud` command throw authentication failed exception even enter the right username\password
It looks some endpoints are wrong.

Steps to reproduce:
1) Run `azure login -e AzureChinaCloud`
2) Then Open IE to access https://aka.ms/deviceloginchina
3) Enter code BCNVMGQU4
4) Enter the username/password
5) the webpage show login successfully.
6) Azure CLI throw authentication failed exception.

Error stack trace:

info:    Executing command login
verbose: Authenticating...
info:    To sign in, use a web browser to open the page https://aka.ms/deviceloginchina. Enter the code BCNVMGQU4 to authenticate.
verbose: code response from AAD is :{"userCode":"BCNVMGQU4","deviceCode":"BAAABAAEAqxyr68extkmmW2YfrQ56e3oXz8-W384wC8x1fMZRDNRghWtk11ij_iu1feLjx3-wGdTpEnT5OyemHsIT50P2e7asMp4TlmAu_FRk-xQB8s6T9CmYvV8OBGernHRk5ByGIAA","verificationUrl":"https://aka.ms/deviceloginchina","expiresIn":900,"interval":5,"message":"To sign in, use a web browser to open the page https://aka.ms/deviceloginchina. Enter the code BCNVMGQU4 to authenticate."}
error:   Authentication failed.
silly:   {
silly:       __frame: {
silly:           name: '__1',
silly:           line: 49,
silly:           file: 'C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\lib\\commands\\login.js',
silly:           prev: undefined,
silly:           calls: 0,
silly:           active: false,
silly:           offset: 64,
silly:           col: 27
silly:       },
silly:       response: {
silly:           body: '{"error":{"code":"AuthenticationFailed","message":"Authentication failed."}}',
silly:           headers: {
silly:               cache-control: 'no-cache',
silly:               pragma: 'no-cache',
silly:               content-type: 'application/json; charset=utf-8',
silly:               expires: '-1',
silly:               www-authenticate: 'Bearer authorization_uri="https://login.windows.net/", error="invalid_token", error_description="Could not find identity for access token."',
silly:               x-ms-failure-cause: 'gateway',
silly:               x-ms-request-id: '7d2e72db-9cbe-4d4b-b1e9-83f6ac60fae1',
silly:               x-ms-correlation-request-id: '7d2e72db-9cbe-4d4b-b1e9-83f6ac60fae1',
silly:               x-ms-routing-request-id: 'JAPANEAST:20160316T113443Z:7d2e72db-9cbe-4d4b-b1e9-83f6ac60fae1',
silly:               strict-transport-security: 'max-age=31536000; includeSubDomains',
silly:               date: 'Wed, 16 Mar 2016 11:34:42 GMT',
silly:               connection: 'close',
silly:               content-length: '76'
silly:           },
silly:           statusCode: 401
silly:       },
silly:       message: 'Authentication failed.',
silly:       statusCode: 401,
silly:       body: { code: 'AuthenticationFailed', message: 'Authentication failed.' },
silly:       rawStack: 'Error: Authentication failed.\n    at C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\azure-arm-resource\\lib\\subscription\\operations\\tenants.js:114:19\n    at retryCallback (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\ms-rest-azure\\node_modules\\ms-rest\\lib\\filters\\exponentialRetryPolicyFilter.js:140:9)\n    at handleRedirect (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\ms-rest-azure\\node_modules\\ms-rest\\lib\\filters\\redirectFilter.js:36:9)\n    at Request._callback (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\ms-rest-azure\\node_modules\\ms-rest\\lib\\requestPipeline.js:125:16)\n    at Request.self.callback (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\request\\request.js:344:22)\n    at emitTwo (events.js:87:13)\n    at Request.emit (events.js:172:7)\n    at Request.<anonymous> (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\request\\request.js:1239:14)\n    at emitOne (events.js:82:20)\n    at Request.emit (events.js:169:7)',
silly:       stack: 'Error: Authentication failed.\n  <<< async stack >>>\n  at __1 (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\lib\\commands\\login.js:113:28)\n  <<< raw stack >>>\n    at C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\azure-arm-resource\\lib\\subscription\\operations\\tenants.js:114:19\n    at retryCallback (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\ms-rest-azure\\node_modules\\ms-rest\\lib\\filters\\exponentialRetryPolicyFilter.js:140:9)\n    at handleRedirect (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\ms-rest-azure\\node_modules\\ms-rest\\lib\\filters\\redirectFilter.js:36:9)\n    at Request._callback (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\ms-rest-azure\\node_modules\\ms-rest\\lib\\requestPipeline.js:125:16)\n    at Request.self.callback (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\request\\request.js:344:22)\n    at emitTwo (events.js:87:13)\n    at Request.emit (events.js:172:7)\n    at Request.<anonymous> (C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI\\node_modules\\request\\request.js:1239:14)\n    at emitOne (events.js:82:20)\n    at Request.emit (events.js:169:7)',
silly:       request: {
silly:           rawResponse: false,
silly:           queryString: {},
silly:           method: 'GET',
silly:           headers: {
silly:               Content-Type: 'application/json; charset=utf-8',
silly:               accept-language: 'en-US',
silly:               x-ms-client-request-id: '8121d680-37e2-4ae0-ba65-7510c4dbd526'
silly:           },
silly:           url: 'https://management.azure.com/tenants?api-version=2015-11-01',
silly:           body: null
silly:       },
silly:       code: 'AuthenticationFailed'
silly:   }
info:    Error information has been recorded to C:\Users\weisun\.azure\azure.err
verbose: Error: Authentication failed.
  <<< async stack >>>
  at __1 (C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\lib\commands\login.js:113:28)
  <<< raw stack >>>
    at C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\node_modules\azure-arm-resource\lib\subscription\operations\tenants.js:114:19
    at retryCallback (C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\node_modules\ms-rest-azure\node_modules\ms-rest\lib\filters\exponentialRetryPolicyFilter.js:140:9)
    at handleRedirect (C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\node_modules\ms-rest-azure\node_modules\ms-rest\lib\filters\redirectFilter.js:36:9)
    at Request._callback (C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\node_modules\ms-rest-azure\node_modules\ms-rest\lib\requestPipeline.js:125:16)
    at Request.self.callback (C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\node_modules\request\request.js:344:22)
    at emitTwo (events.js:87:13)
    at Request.emit (events.js:172:7)
    at Request.<anonymous> (C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\node_modules\request\request.js:1239:14)
    at emitOne (events.js:82:20)
    at Request.emit (events.js:169:7)
error:   login command failed

**Comment by amarzavery** [comment]
@catweisun - Thanks for reporting this. We shall fix this in the upcoming release.
The following log in the line looks problematic. It shouldn't be using the Azure Public Cloud URI.
```
silly: www-authenticate: 'Bearer authorization_uri="https://login.windows.net/", error="invalid_token", error_description="Could not find identity for access token."',
 silly: x-ms-failure-cause: 'gateway',
```

**Comment by amarzavery** [comment]
@catweisun - This issue has been fixed via the 0.9.18 hotfix. The hotfix is available only via npm.
Please follow the instructions as follows:
1.     uninstall the CLI via MSI
2.     open an admin cmd prompt.
3.	npm cache clear –g
4.	delete the .streamline directory from your home profile (C:\Users\<username>\.streamline)
5.	npm install azure-cli –g

This should install the latest version 0.9.18 on your machine. 

Try the login command again for the AzureChinaCloud environment and mostly you should be good to go.

**Comment by catweisun** [comment]
Thanks, it works fine now
